awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsUnsampledOnlySpanProcessor.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.opentelemetry.javaagent.providers;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+/**
+ * {@link SpanProcessor} that only exports unsampled spans in a batch via a delegated @{link
+ * BatchSpanProcessor}. The processor also adds an attribute to each processed span to indicate that
+ * it was sampled or not.
+ */
+final class AwsUnsampledOnlySpanProcessor implements SpanProcessor {
+
+  private final SpanProcessor delegate;
+
+  AwsUnsampledOnlySpanProcessor(SpanProcessor delegate) {
+    this.delegate = delegate;
+  }
+
+  public static AwsUnsampledOnlySpanProcessorBuilder builder() {
+    return new AwsUnsampledOnlySpanProcessorBuilder();
+  }
+
+  @Override
+  public void onStart(Context parentContext, ReadWriteSpan span) {
+    if (span.getSpanContext().isSampled()) {
+      span.setAttribute(AwsAttributeKeys.AWS_TRACE_FLAG_SAMPLED, true);
+    } else {
+      span.setAttribute(AwsAttributeKeys.AWS_TRACE_FLAG_SAMPLED, false);
```

Comment:
We don't need FLAG_SAMPLED=false if span is sampled, it can save span size a little bit

Reply:
I see your point and I can make the change to align with current implementations in Python and JavaScript.

However, as a boolean span attribute, `aws.trace.flag.sampled` should always be present for consistency. Without `aws.trace.flag.sampled=false`, it could be unclear whether the span was unsampled or if there was a malfunction in the SpanProcessor and the span was actually sampled. Ensuring this attribute is always present would improve clarity. Let me know if you agree, and I can assist with implementing this change in Python and JavaScript.

Reply:
It is ok to have no this attribute since by default the span in backend should be sampled.
Having this attribute actually is specific to a CloudWatch feature, it means backend can add true if want to keep consistency for user experience.

Reply:
ok. made the change.